python/Ganga/new_tests/GPI/Loading/LazyLoadingTest.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import absolute_import
+
+from ..GangaUnitTest import GangaUnitTest
+
+class LazyLoadingTest(GangaUnitTest):
+
+    def setUp(self):
+        """Make sure that the Job object isn't destroyed between tests"""
+        super(LazyLoadingTest, self).setUp()
+        from Ganga.Utility.Config import setConfigOption
+        setConfigOption('TestingFramework', 'AutoCleanup', 'False')
+
+    def test_a_JobConstruction(self):
+        """ First construct the Job object (singular)"""
+        from Ganga.Utility.Config import getConfig
+        self.assertFalse(getConfig('TestingFramework')['AutoCleanup'])
+
+        from Ganga.GPI import Job, jobs
+        j=Job()
+        self.assertTrue(len(jobs) == 1)
+
+    def test_b_JobNotLoaded(self):
+        """ Second get the job and check that getting it via jobs doesn't cause it to be loaded"""
+        from Ganga.GPI import jobs
+
+        self.assertTrue(len(jobs) == 1)
```

Comment:
This could be accomplished with some well placed config commands in one of the setUp or setUpClass I think. I'll look into this.
I'd be tempted to just add a hard remove command into a tearDownClass which uses os level commands to remove the files from disk after Ganga has shutdown.